diff --git a/networking/udhcp/common.h b/networking/udhcp/common.h
index 5882238..3a5a4fb 100644
--- a/networking/udhcp/common.h
+++ b/networking/udhcp/common.h
@@ -150,7 +150,7 @@ struct dhcp_scan_state {
 //#define DHCP_IP_TTL           0x17
 //#define DHCP_MTU              0x1a
 //#define DHCP_BROADCAST        0x1c
-//#define DHCP_ROUTES           0x21
+#define DHCP_ROUTES           0x21
 //#define DHCP_NIS_DOMAIN       0x28
 //#define DHCP_NIS_SERVER       0x29
 //#define DHCP_NTP_SERVER       0x2a
@@ -176,13 +176,13 @@ struct dhcp_scan_state {
 //#define DHCP_TCODE            0x65 /* 101: RFC 4833. Reference to the TZ database string */
 //#define DHCP_DOMAIN_SEARCH    0x77 /* 119: RFC 3397. set of ASCIZ string, DNS-style compressed */
 //#define DHCP_SIP_SERVERS      0x78 /* 120: RFC 3361. flag byte, then: 0: domain names, 1: IP addrs */
-//#define DHCP_STATIC_ROUTES    0x79 /* 121: RFC 3442. (mask,ip,router) tuples */
+#define DHCP_STATIC_ROUTES    0x79 /* 121: RFC 3442. (mask,ip,router) tuples */
 //#define DHCP_VLAN_ID          0x84 /* 132: 802.1P VLAN ID */
 //#define DHCP_VLAN_PRIORITY    0x85 /* 133: 802.1Q VLAN priority */
 //#define DHCP_PXE_CONF_FILE    0xd1 /* 209: RFC 5071 Configuration file */
 //#define DHCP_PXE_PATH_PREFIX  0xd2 /* 210: RFC 5071 Path prefix */
 //#define DHCP_REBOOT_TIME      0xd3 /* 211: RFC 5071 Reboot time */
-//#define DHCP_MS_STATIC_ROUTES 0xf9 /* 249: Microsoft's pre-RFC 3442 code for 0x79? */
+#define DHCP_MS_STATIC_ROUTES 0xf9 /* 249: Microsoft's pre-RFC 3442 code for 0x79? */
 //#define DHCP_WPAD             0xfc /* 252: MSIE's Web Proxy Autodiscovery Protocol */
 #define DHCP_END                0xff /* 255: */
 
