diff -aur old/arch/arm/Kconfig new/arch/arm/Kconfig
--- old/arch/arm/Kconfig	2022-09-24 09:39:00.120731915 +0800
+++ new/arch/arm/Kconfig	2022-09-24 09:37:43.668876222 +0800
@@ -609,6 +609,8 @@
 # Kconfigs may be included either alphabetically (according to the
 # plat- suffix) or along side the corresponding mach-* source.
 #
+source "arch/arm/mach-triductor/Kconfig"
+
 source "arch/arm/mach-actions/Kconfig"
 
 source "arch/arm/mach-alpine/Kconfig"
