diff -aur old/arch/arm/Makefile new/arch/arm/Makefile
--- old/arch/arm/Makefile	2022-08-25 17:38:24.000000000 +0800
+++ new/arch/arm/Makefile	2022-09-29 19:37:24.482943240 +0800
@@ -155,6 +155,7 @@
 
 # Machine directory name.  This list is sorted alphanumerically
 # by CONFIG_* macro name.
+machine-$(CONFIG_ARCH_TRIDUCTOR)	+= triductor
 machine-$(CONFIG_ARCH_ACTIONS)		+= actions
 machine-$(CONFIG_ARCH_ALPINE)		+= alpine
 machine-$(CONFIG_ARCH_ARTPEC)		+= artpec
