diff -uprN a/src/ap/hw_features.c b/src/ap/hw_features.c
--- a/src/ap/hw_features.c	2023-08-23 13:52:10.349027503 +0800
+++ b/src/ap/hw_features.c	2023-09-12 12:19:42.476668923 +0800
@@ -687,7 +687,9 @@ static int ieee80211ax_supported_he_capa
 
 int hostapd_check_ht_capab(struct hostapd_iface *iface)
 {
+#ifndef CONFIG_CONNECTIVITY_PATCH
 	int ret;
+#endif
 
 	if (is_6ghz_freq(iface->freq))
 		return 0;
@@ -714,9 +716,11 @@ int hostapd_check_ht_capab(struct hostap
 	    !ieee80211ac_supported_vht_capab(iface))
 		return -1;
 #endif /* CONFIG_IEEE80211AC */
+#ifndef CONFIG_CONNECTIVITY_PATCH
 	ret = ieee80211n_check_40mhz(iface);
 	if (ret)
 		return ret;
+#endif /* CONFIG_CONNECTIVITY_PATCH */
 	if (!ieee80211n_allowed_ht40_channel_pair(iface))
 		return -1;
 
