diff -aur old/templates/lxc-busybox.in new/templates/lxc-busybox.in
--- old/templates/lxc-busybox.in	2022-02-02 12:26:37.000000000 +0800
+++ new/templates/lxc-busybox.in	2023-04-17 09:28:25.239688584 +0800
@@ -23,7 +23,7 @@
 LXC_MAPPED_UID=
 LXC_MAPPED_GID=
 
-BUSYBOX_EXE=$(command -v busybox)
+BUSYBOX_EXE='/usr/share/lxc/containers/busybox'
 
 # Make sure the usual locations are in PATH
 export PATH="$PATH:/usr/sbin:/usr/bin:/sbin:/bin"
